plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java

```diff
@@ -205,4 +211,27 @@ public String[] getFuncAliases() {
         return funcAliases;
     }
 
+    /**
+     * Represents a formatted table in an SQL query.
+     * This class is used to format the table name according to the specified SQL dialect.
+     * It is necessary because `net.sf.jsqlparser.schema.Table` only supports a dot as a separator.
+     */
+    // TODO: Remove this class when https://github.com/dbeaver/pro/issues/3140 will be resolved
+    private static class FormattedTable extends Table {
+        private final SQLDialect sqlDialect;
+
+        public FormattedTable(Table table, SQLDialect sqlDialect) {
+            super(table.getDatabase(), table.getSchemaName(), table.getName());
+            this.sqlDialect = sqlDialect;
+        }
+
+        @Override
+        public String toString() {
+            String databaseName = getDatabase().getDatabaseName() != null ? getDatabase().getDatabaseName() + sqlDialect.getCatalogSeparator() : "";
+            String schemaName = getSchemaName() != null ? getSchemaName() + sqlDialect.getStructSeparator() : "";
+            String fullyQualifiedTableName = databaseName + schemaName + getName();
+            String alias = getAlias() != null ? getAlias().toString() : "";
+            return fullyQualifiedTableName + alias;
```

Comment:
Looks weird. name + alias without space between?

Reply:
Perhaps it is better to use StringBuilder here.

Reply:
`net.sf.jsqlparser.expression.Alias#toString` already adds a space before the alias. Do we really consider this a bottleneck that justifies sacrificing readability for performance?

Reply:
Do we really consider this a bottleneck that justifies sacrificing readability for performance?

Reply:
No it is just weird looking code. Relying on undocumented feature of 3rd party library doesn't sound very good.

Performance here doesn't really matter, it is about readability.

Reply:
Yeah, agree.

I tried to do that using SB, but I found the result even less readable

```
@Override
public String toString() {
    StringBuilder fullyQualifiedTableName = new StringBuilder();
    
    if (getDatabase().getDatabaseName() != null) {
        fullyQualifiedTableName.append(getDatabase().getDatabaseName())
                               .append(sqlDialect.getCatalogSeparator());
    }
    if (getSchemaName() != null) {
        fullyQualifiedTableName.append(getSchemaName())
                               .append(sqlDialect.getStructSeparator());
    }
    
    fullyQualifiedTableName.append(getName());
    
    if (getAlias() != null) {
        fullyQualifiedTableName.append(getAlias().toString());
    }
    
    return fullyQualifiedTableName.toString();
}

```